server/src/main/java/com/hedera/block/server/pbj/PbjBlockAccessServiceProxy.java

```diff
@@ -0,0 +1,157 @@
+                        .mapRequest(bytes -> parseSingleBlockRequest(bytes, options))
+                        .method(this::singleBlock)
+                        .mapResponse(reply -> createSingleBlockResponse(reply, options))
+                        .respondTo(replies)
+                        .build();
+            };
+        } catch (Exception e) {
+            replies.onError(e);
+            return Pipelines.noop();
+        }
+    }
+
+    SingleBlockResponse singleBlock(SingleBlockRequest singleBlockRequest) {
+
+        LOGGER.log(DEBUG, "Executing Unary singleBlock gRPC method");
+
+        if (serviceStatus.isRunning()) {
+            final long blockNumber = singleBlockRequest.blockNumber();
+            try {
+                final Optional<Block> blockOpt = blockReader.read(blockNumber);
+                if (blockOpt.isPresent()) {
+                    LOGGER.log(DEBUG, "Successfully returning block number: {0}", blockNumber);
+                    metricsService.get(SingleBlocksRetrieved).increment();
+
+                    return SingleBlockResponse.newBuilder()
+                            .status(SingleBlockResponseCode.READ_BLOCK_SUCCESS)
+                            .block(blockOpt.get())
+                            .build();
+                } else {
+                    LOGGER.log(DEBUG, "Block number {0} not found", blockNumber);
+                    metricsService.get(SingleBlocksNotFound).increment();
+
+                    return SingleBlockResponse.newBuilder()
+                            .status(SingleBlockResponseCode.READ_BLOCK_NOT_FOUND)
+                            .build();
+                }
+            } catch (IOException e) {
+                LOGGER.log(ERROR, "Error reading block number: {0}", blockNumber);
+
+                return SingleBlockResponse.newBuilder()
+                        .status(SingleBlockResponseCode.READ_BLOCK_NOT_AVAILABLE)
+                        .build();
+            } catch (ParseException e) {
+                LOGGER.log(ERROR, "Error parsing block number: {0}", blockNumber);
+
+                return SingleBlockResponse.newBuilder()
+                        .status(SingleBlockResponseCode.READ_BLOCK_NOT_AVAILABLE)
+                        .build();
+            }
+        } else {
+            LOGGER.log(ERROR, "Unary singleBlock gRPC method is not currently running");
+
+            return SingleBlockResponse.newBuilder()
+                    .status(SingleBlockResponseCode.READ_BLOCK_NOT_AVAILABLE)
+                    .build();
+        }
+    }
+
+    @NonNull
+    private SingleBlockRequest parseSingleBlockRequest(
+            @NonNull final Bytes message, @NonNull final RequestOptions options) throws ParseException {
+        // Copying bytes to avoid using references passed from Helidon
+        return SingleBlockRequest.PROTOBUF.parse(Bytes.wrap(message.toByteArray()));
+    }
+
+    @NonNull
```

Comment:
are we still using @NonNull annotation on this cases? I thought we are only using it on parameters, constructors and such.